### ISSUE TITLE
Build failed on Windows

### ISSUE THREAD

**Issue by beachwalker** [gwaldron/osgearth#814]
I tried to compile OsgEarth 2.8 on Windows 7 using Visual Studio 2013. During the build the an error occured when building osgEarthFeatures. Everything else (except dependent upon osgEarthFeatures lib) seems to compile properly.
It looks like the cmd shell can't execute the script (MSB 6006, Exit Code 3).

1>  Building Custom Rule D:/Projects/Thirdparty/source/osgearth-2.8/src/osgEarthFeatures/CMakeLists.txt
1>  CMake does not need to re-run because D:\Projects\Thirdparty\build\x64\Release\osgearth\src\osgEarthFeatures\CMakeFiles\generate.stamp is up-to-date.
1>  Running C++ protocol buffer compiler on vector_tile.proto
1>  Das System kann den angegebenen Pfad nicht finden. (translation: The system is unable to find the given path.)

**Comment by beachwalker** [comment]
I think I have found a issue during compilation against protocol buffers. Which version do you use exeactly for compilation? 2.7, 3.0, 3.1?

**Comment by beachwalker** [comment]
This is the output of the compile result:

```
"D:\Projects\Thirdparty\build\x64\Release\osgearth\install.vcxproj" (Standardzi
el) (1) ->
"D:\Projects\Thirdparty\build\x64\Release\osgearth\ALL_BUILD.vcxproj" (Standard
ziel) (2) ->
"D:\Projects\Thirdparty\build\x64\Release\osgearth\src\applications\osgearth_3p
v\application_osgearth_3pv.vcxproj" (Standardziel) (4) ->
"D:\Projects\Thirdparty\build\x64\Release\osgearth\src\osgEarthAnnotation\osgEa
rthAnnotation.vcxproj" (Standardziel) (6) ->
"D:\Projects\Thirdparty\build\x64\Release\osgearth\src\osgEarthFeatures\osgEart
hFeatures.vcxproj" (Standardziel) (7) ->
(Link Ziel) ->
  MVT.obj : error LNK2019: Verweis auf nicht aufgel÷stes externes Symbol ""clas
s std::basic_string<char,struct std::char_traits<char>,class std::allocator<cha
r> > const * const google::protobuf::internal::empty_string_" (?empty_string_@i
nternal@protobuf@google@@3PEBV?$basic_string@DU?$char_traits@D@std@@V?$allocato
r@D@2@@std@@EB)" in Funktion ""class std::basic_string<char,struct std::char_tr
aits<char>,class std::allocator<char> > const & __cdecl google::protobuf::inter
nal::GetEmptyStringAlreadyInited(void)" (?GetEmptyStringAlreadyInited@internal@
protobuf@google@@YAAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@
std@@XZ)". [D:\Projects\Thirdparty\build\x64\Release\osgearth\src\osgEarthFeatu
res\osgEarthFeatures.vcxproj]
  vector_tile.pb.obj : error LNK2001: Nicht aufgel÷stes externes Symbol ""class
 std::basic_string<char,struct std::char_traits<char>,class std::allocator<char
> > const * const google::protobuf::internal::empty_string_" (?empty_string_@in
ternal@protobuf@google@@3PEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator
@D@2@@std@@EB)". [D:\Projects\Thirdparty\build\x64\Release\osgearth\src\osgEart
hFeatures\osgEarthFeatures.vcxproj]
  vector_tile.pb.obj : error LNK2019: Verweis auf nicht aufgel÷stes externes Sy
mbol ""__int64 google::protobuf::internal::empty_string_once_init_" (?empty_str
ing_once_init_@internal@protobuf@google@@3_JA)" in Funktion ""class std::basic_
string<char,struct std::char_traits<char>,class std::allocator<char> > const &
__cdecl google::protobuf::internal::GetEmptyString(void)" (?GetEmptyString@inte
rnal@protobuf@google@@YAAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@
D@2@@std@@XZ)". [D:\Projects\Thirdparty\build\x64\Release\osgearth\src\osgEarth
Features\osgEarthFeatures.vcxproj]
  D:\Projects\Thirdparty\build\x64\Release\osgearth\lib\Release\osgEarthFeature
s.dll : fatal error LNK1120: 2 nicht aufgel÷ste Externe [D:\Projects\Thirdparty
\build\x64\Release\osgearth\src\osgEarthFeatures\osgEarthFeatures.vcxproj]
```

**Comment by beachwalker** [comment]
I'm building x64 using Visual Studo 2013. All dependencies are also built for x64 and linked accordingly. OsgEarth works fine when building WITHOUT Protocol Buffers but now I want to test the new features comming with rex and Protocol Buffers.

**Comment by beachwalker** [comment]
My presets for CMake related to protobuf:

```
set(PROTOBUF_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/../../../../install/x64/Release/protobuf/include" CACHE PATH "")
set(PROTOBUF_LIBRARY "${CMAKE_SOURCE_DIR}/../../../../install/x64/Release/protobuf/lib/libprotobuf.lib" CACHE PATH "")
set(PROTOBUF_LIBRARY_DEBUG "${CMAKE_SOURCE_DIR}/../../../../install/x64/Debug/protobuf/lib/libprotobuf.lib" CACHE PATH "")
set(PROTOBUF_PROTOC_EXECUTABLE "${CMAKE_SOURCE_DIR}/../../../../install/x64/Release/protobuf/bin/protoc.exe" CACHE PATH "")
set(PROTOBUF_PROTOC_LIBRARY "${CMAKE_SOURCE_DIR}/../../../../install/x64/Release/protobuf/lib/libprotoc.lib" CACHE PATH "")
set(PROTOBUF_PROTOC_LIBRARY_DEBUG "${CMAKE_SOURCE_DIR}/../../../../install/x64/Debug/protobuf/lib/libprotoc.lib" CACHE PATH "")
set(PROTOBUF_LITE_LIBRARY "${CMAKE_SOURCE_DIR}/../../../../install/x64/Release/protobuf/lib/libprotobuf-lite.lib" CACHE PATH "")
set(PROTOBUF_LITE_LIBRARY_DEBUG "${CMAKE_SOURCE_DIR}/../../../../install/x64/Debug/protobuf/lib/libprotobuf-lite.lib" CACHE PATH "")

```

**Comment by jasonbeverage** [comment]
I'm currently building against protocol buffers 2.6.1 on windows.

**Comment by beachwalker** [comment]
Hi Jason,

thank you for the info. This was exactly the info I needed to solve the problem. 2.6.1 was the last version that worked. 2.6.1 is somewhat outdated and did not use CMake (provided vs files, but only for Win32 configuration etc.). Google changed this and it is much easier to build it now using different compilers and platforms.

Beyond having to build against an outdated version that is more difficult to build I recommend to upgrade osgEarth to work against 3.x (cause 2.6.x branches are no longer maintained), too. They have been released a while ago.
